addon_service/oauth1/views.py

```diff
@@ -5,26 +5,26 @@
 
 from addon_service.authorized_storage_account.models import AuthorizedStorageAccount
 from addon_service.common.known_imps import AddonImpNumbers
+from addon_service.oauth1.models import OAuth1TemporaryCredentials
 from addon_service.oauth1.utils import get_access_token
 from addon_service.oauth_utlis import update_external_account_id
-from addon_service.osf_models.fields import decrypt_string
 
 
 def oauth1_callback_view(request):
-    oauth_token = request.GET["oauth_token"]
+    temporary_oauth_token = request.GET["oauth_token"]
     oauth_verifier = request.GET["oauth_verifier"]
 
-    pk = decrypt_string(request.session.get("oauth1a_account_id"))
-    del request.session["oauth1a_account_id"]
-
-    account = AuthorizedStorageAccount.objects.get(pk=pk)
-
+    account = AuthorizedStorageAccount.objects.get(
+        _temporary_oauth1_credentials__in=OAuth1TemporaryCredentials.objects.filter_by_oauth1_temporary_token(
+            temporary_oauth_token
+        )
+    )
```

Comment:
this won't work as token from `request.GET["oauth_token"]` is different to `temporary_oauth_token` obtained before forming auth url. I tested it. If this wasn't a case, I would've implemented it in the same way as you

Reply:
hmm ok that seems to agree with [example code in zotero docs](https://www.zotero.org/support/dev/web_api/v3/oauth#example_php), which uses the new token for the next request...
```
        //set the token we got back from the provider and the secret we saved previously for the exchange.
        $oauth->setToken($_GET['oauth_token'], $request_token_info['oauth_token_secret']);
```

...but i've been confused because that seems to disagree with oauth1, both [rfc-5849](https://www.rfc-editor.org/rfc/rfc5849#section-2.2)
```
   [...] The server constructs the request URI by adding the
   following REQUIRED parameters to the callback URI query component:

   oauth_token
         The temporary credentials identifier received from the client.
   oauth_verifier
         The verification code.
```
and [1.0a](https://oauth.net/core/1.0a/#rfc.section.6.2.3)
```
   [...], and directs the User's web browser to that URL with the following parameters added:

    oauth_token:
        The Request Token the User authorized or denied. 
    oauth_verifier:
        The verification code. 
```
so i'd assumed the temporary token was meant to be shared across all the requests in a given auth attempt -- implementing zotero's way (with account id in the session) seems fine, since it supports services whether they use the temporary token once or twice

Reply:
I guess it indeed was, Zotero's just an exemption and the only OAuth1.0a service we have (as of now)